store/tikv/kv.go

```diff
@@ -234,8 +235,38 @@ func (s *tikvStore) IsLatchEnabled() bool {
 	return s.txnLatches != nil
 }
 
-func (s *tikvStore) EtcdAddrs() []string {
-	return s.etcdAddrs
+func (s *tikvStore) EtcdAddrs() ([]string, error) {
+	if s.etcdAddrs == nil {
+		return nil, nil
+	}
+	ctx := context.Background()
+	bo := NewBackoffer(ctx, GetMemberInfoBackoff)
+	etcdAddrs := make([]string, 0)
+	pdClient := s.pdClient
+	if pdClient == nil {
+		return nil, errors.New("Etcd client not found")
+	}
+	for {
+		members, err := pdClient.GetMemberInfo(ctx)
+		if err != nil {
+			err := bo.Backoff(BoRegionMiss, err)
+			if err != nil {
+				return nil, err
+			}
+			continue
+		}
+		for _, member := range members {
+			if len(member.ClientUrls) > 0 {
+				u, err := url.Parse(member.ClientUrls[0])
+				if err != nil {
+					logutil.BgLogger().Error("fail to parse client url from pd members", zap.String("client_url", member.ClientUrls[0]), zap.Error(err))
+					continue
```

Comment:
Why not return here?

Reply:
Becaue I think that parer this member's url have error. but maybe parse the next member url is success

Reply:
Maybe return error is better, parse URL error is not expected.